dav1d-sys/src/lib.rs

```diff
@@ -3,7 +3,664 @@
+    pub frame_hdr_ref: *mut Dav1dRef,
+    pub seq_hdr_ref: *mut Dav1dRef,
+    pub content_light_ref: *mut Dav1dRef,
+    pub mastering_display_ref: *mut Dav1dRef,
+    pub itut_t35_ref: *mut Dav1dRef,
+    pub reserved_ref: [usize; 4usize],
+    pub ref_: *mut Dav1dRef,
+    pub allocator_data: *mut ::std::os::raw::c_void,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Dav1dPicAllocator {
+    pub cookie: *mut ::std::os::raw::c_void,
+    pub alloc_picture_callback: ::std::option::Option<
+        unsafe extern "C" fn(
+            pic: *mut Dav1dPicture,
+            cookie: *mut ::std::os::raw::c_void,
+        ) -> ::std::os::raw::c_int,
+    >,
+    pub release_picture_callback: ::std::option::Option<
+        unsafe extern "C" fn(pic: *mut Dav1dPicture, cookie: *mut ::std::os::raw::c_void),
+    >,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Dav1dData {
+    pub data: *const u8,
+    pub sz: usize,
+    pub ref_: *mut Dav1dRef,
+    pub m: Dav1dDataProps,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Dav1dContext {
+    _unused: [u8; 0],
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Dav1dLogger {
+    pub cookie: *mut ::std::os::raw::c_void,
+    pub callback: *mut ::std::os::raw::c_void,
+    // FIXME: Use the following once std::ffi::VaList is stable
+    // ::std::option::Option<
+    //     unsafe extern "C" fn(
+    //         cookie: *mut ::std::os::raw::c_void,
+    //         format: *const ::std::os::raw::c_char,
+    //         ap: *mut ::std::ffi::VaList,
+    //     ),
+    //  >,
+}
+
+pub const Dav1dInloopFilterType_DAV1D_INLOOPFILTER_NONE: Dav1dInloopFilterType = 0;
+pub const Dav1dInloopFilterType_DAV1D_INLOOPFILTER_DEBLOCK: Dav1dInloopFilterType = 1;
+pub const Dav1dInloopFilterType_DAV1D_INLOOPFILTER_CDEF: Dav1dInloopFilterType = 2;
+pub const Dav1dInloopFilterType_DAV1D_INLOOPFILTER_RESTORATION: Dav1dInloopFilterType = 4;
+pub const Dav1dInloopFilterType_DAV1D_INLOOPFILTER_ALL: Dav1dInloopFilterType = 7;
+pub type Dav1dInloopFilterType = ::std::os::raw::c_uint;
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Dav1dSettings {
+    pub n_threads: ::std::os::raw::c_int,
+    pub max_frame_delay: ::std::os::raw::c_int,
+    pub apply_grain: ::std::os::raw::c_int,
+    pub operating_point: ::std::os::raw::c_int,
+    pub all_layers: ::std::os::raw::c_int,
+    pub frame_size_limit: ::std::os::raw::c_uint,
+    pub allocator: Dav1dPicAllocator,
+    pub logger: Dav1dLogger,
+    pub strict_std_compliance: ::std::os::raw::c_int,
+    pub output_invisible_frames: ::std::os::raw::c_int,
+    pub inloop_filters: Dav1dInloopFilterType,
+    pub reserved: [u8; 20usize],
+}
+
+pub const Dav1dEventFlags_DAV1D_EVENT_FLAG_NEW_SEQUENCE: Dav1dEventFlags = 1;
+pub const Dav1dEventFlags_DAV1D_EVENT_FLAG_NEW_OP_PARAMS_INFO: Dav1dEventFlags = 2;
+pub type Dav1dEventFlags = ::std::os::raw::c_uint;
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Dav1dRef {
```

Comment:
This should probably just be a `#[repr(transparent)]` struct around `c_void`, and especially the way it was generated by bindgen here is actually wrong